runtime/drivers/s3/api.go

```diff
@@ -2,54 +2,23 @@ package s3
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"net/http"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 	"gocloud.dev/blob"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 func (c *Connection) ListBuckets(ctx context.Context) ([]string, error) {
-	creds, err := c.newCredentials()
+	client, err := getS3Client(ctx, *c.config, "")
```

Comment:
These dereferencing arg passes `*c.config` look wrong – I agree with Anshul that we should be consistent and either use values or pointers everywhere, not mix and match.